src/store/eventing/middleware.ts

```diff
@@ -0,0 +1,28 @@
+import noop from 'lodash/noop';
+import { Action, Dispatch, Middleware, MiddlewareAPI } from '@reduxjs/toolkit';
+import { saveAnnotationAction } from '../annotations/actions';
+import { handleCreateErrorEvents, handleCreatePendingEvents, handleCreateSuccessEvents } from './create';
+import { EventHandler, EventHandlerMap } from '../../@types';
+
+// Array of event handlers based on redux action. To add handling for new events add an entry keyed by action
+const eventHandlers: EventHandlerMap = {
+    [saveAnnotationAction.pending.toString()]: handleCreatePendingEvents,
+    [saveAnnotationAction.fulfilled.toString()]: handleCreateSuccessEvents,
+    [saveAnnotationAction.rejected.toString()]: handleCreateErrorEvents,
+};
+
+const getEventingMiddleware: Middleware = (handlers?: EventHandlerMap = eventHandlers) => (store: MiddlewareAPI) => (
+    next: Dispatch,
+) => (action: Action): Action => {
+    const { type } = action;
+    const prevState = store.getState();
+    const result = next(action);
+    const nextState = store.getState();
```

Comment:
It's not super clear why this order is needed, but I assume it's so the action can modify the state. Should we call it out in a comment?